datastore/sync_entity.go

```diff
@@ -159,7 +165,7 @@ func (dynamo *Dynamo) InsertSyncEntity(entity *SyncEntity) (bool, error) {
 		return false, fmt.Errorf("error building expression to insert sync entity: %w", err)
 	}
 
-	if entity.ClientDefinedUniqueTag != nil {
+	if entity.ClientDefinedUniqueTag != nil && *entity.DataType != historyTypeID {
```

Comment:
Please update the comment above this function to mention the history type case does not use tag item.